HeterogeneousCore/AlpakaCore/interface/alpaka/typelookup.h

```diff
@@ -0,0 +1,29 @@
+#ifndef HeterogeneousCore_AlpakaCore_interface_alpaka_typelookup_h
+#define HeterogeneousCore_AlpakaCore_interface_alpaka_typelookup_h
+
+#include "FWCore/Utilities/interface/typelookup.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ESDeviceProduct.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+
+// TODO: this place is suboptimal given that the framework's
+// typelookup.h is in FWCore/Utilities (i.e. independent of the
+// framework itself). The ESDeviceProduct should be in the same
+// package.
```

Comment:
This file is one of those which should not really be in `HeterogeneousCore/AlpakaCore` because of the package depending on `FWCore/Framework`. The code here depends only on Alpaka and `FWCore/Utilities`, and is specific to CMS (so would not really fit in `HeterogeneousCore/AlpakaInterface`).